tests/compiletest.rs

```diff
@@ -156,9 +163,10 @@ fn main() -> Result<()> {
     // Let the tests know where to store temp files (they might run for a different target, which can make this hard to find).
     env::set_var("MIRI_TEMP", env::temp_dir());
 
-    ui(Mode::Pass, "tests/pass")?;
-    ui(Mode::Panic, "tests/panic")?;
-    ui(Mode::Fail, "tests/fail")?;
+    ui(Mode::Pass, "tests/pass", false)?;
```

Comment:
Magic booleans make this quite hard to read, please either add comments like
```suggestion
    ui(Mode::Pass, "tests/pass", /* with_dependencies */ false)?;
```
or make this a dedicated enum.

Reply:
😆 I keep forgetting this as rust-analyzer fills this in for me